### PR TITLE
refactor: move builds to use datastore

### PR DESCRIPTION
**Pull request by crowlKats**

### CODE REVIEW

File: api/builds/get.ts [hunk 1]
```diff
@@ -35,18 +36,7 @@ export async function handler(
     });
   }
 
-  try {
-    new Bson.ObjectId(id);
-  } catch (_err: unknown) {
```
Comment (bartlomieju) [inline api/builds/get.ts:40]: Why is this bit removed?

File: utils/test_utils.ts [hunk 2]
```diff
@@ -220,8 +220,23 @@ export async function cleanupDatabase(
...
+      }));
+
+      for await (
+        const _ of datastore.db.commit(mutations, {
+          transactional: false,
+        })
+      ) {
+        //
+      }
+    })(),
```
Comment (bartlomieju) [inline utils/test_utils.ts:239]: Shouldn't be wrapped in `try/catch`?

**Review by bartlomieju — COMMENTED** [review]
Looks good, just two questions

File: api/builds/get.ts (hunk 1 above)
Comment (crowlKats) [reply api/builds/get.ts:40]: BSON ids are only used by mongo. Datastore doesnt have automatic ids, so instead we generate uuids. I guess I could do a check for it being either valid BSON or uuid if thats preferrable, but i dont see too much value in differentiating between invalid build id or just build not found just depending on the data format passed.

File: utils/test_utils.ts (hunk 2 above)
Comment (crowlKats) [reply utils/test_utils.ts:239]: its just for testing; if this fails for some reason, then something would be wrong with either the db or tests, so throwing is ok

File: api/builds/get.ts (hunk 1 above)
Comment (bartlomieju) [reply api/builds/get.ts:40]: Okay, as long as it doesn't break previous data it's fine

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**